packages/app-builder-lib/src/macPackager.ts

```diff
@@ -91,6 +92,42 @@ export default class MacPackager extends PlatformPackager<MacConfiguration> {
     }
   }
 
+  protected async doPack(outDir: string, appOutDir: string, platformName: ElectronPlatformName, arch: Arch, platformSpecificBuildOptions: MacConfiguration, targets: Array<Target>): Promise<any> {
+    switch (arch) {
+      default: {
+        return super.doPack(outDir, appOutDir, this.platform.nodeName as ElectronPlatformName, arch, this.platformSpecificBuildOptions, targets);
+      }
+      case Arch.universal: {
+        const x64Arch = Arch.x64;
+        const x64AppOutDir = appOutDir + '-' + Arch[x64Arch];
+        await super.doPack(outDir, x64AppOutDir, platformName, x64Arch, platformSpecificBuildOptions, targets, false);
+        const arm64Arch = Arch.arm64;
+        const arm64AppOutPath = appOutDir + '-' + Arch[arm64Arch];
+        await super.doPack(outDir, arm64AppOutPath, platformName, arm64Arch, platformSpecificBuildOptions, targets, false);
+        const framework = this.info.framework
+        log.info({
+          platform: platformName,
+          arch: Arch[arch],
+          [`${framework.name}`]: framework.version,
+          appOutDir: log.filePath(appOutDir),
+        }, `packaging`)
+        const appFile = `${this.appInfo.productFilename}.app`;
+        const { makeUniversalApp } = require('@electron/universal');
+        await makeUniversalApp({
+          x64AppPath: path.join(x64AppOutDir, appFile),
+          arm64AppPath: path.join(arm64AppOutPath, appFile),
+          outAppPath: path.join(appOutDir, appFile),
+          force: true
+        });
+        const rmdir = promisify(require('fs').rmdir);
+        await rmdir(x64AppOutDir, { recursive: true });
+        await rmdir(arm64AppOutPath, { recursive: true });
+        await this.doSignAfterPack(outDir, appOutDir, platformName, arch, platformSpecificBuildOptions, targets);
```

Comment:
Minor and probably pedantic, but if we were to consolidate this logic into a `map` it might be less lengthy of a `doPack` override.
```
const appFile = `${this.appInfo.productFilename}.app`;
const [x64AppPath, arm64AppPath] = [Arch.x64, Arch.arm64].map(async arch => {
  const destDir = appOutDir + '-' + Arch[arch];
  await super.doPack(outDir, destDir, platformName, arch, platformSpecificBuildOptions, targets, false);
  return path.join(destDir, appFile)
})
const { version, name } = this.info.framework
log.info({
  platform: platformName,
  arch: Arch[arch],
  [name]: version,
  appOutDir: log.filePath(appOutDir),
}, `packaging`)
const { makeUniversalApp } = require('@electron/universal');
await makeUniversalApp({ x64AppPath, arm64AppPath, outAppPath: path.join(appOutDir, appFile), force: true });

const rmdir = promisify(require('fs').rmdir);
archOutput.forEach(async dir => { await rmdir(dir, { recursive: true }) });
await this.doSignAfterPack(outDir, appOutDir, platformName, arch, platformSpecificBuildOptions, targets);
```

Regardless, this is fantastic work! Everything checks out locally for me as well.